src/Build/Instance/TaskRegistry.cs

```diff
@@ -471,6 +472,12 @@ internal RegisteredTaskRecord GetTaskRegistrationRecord
             retrievedFromCache = false;
             RegisteredTaskIdentity taskIdentity = new RegisteredTaskIdentity(taskName, taskIdentityParameters);
 
+            // was this task overridden?
+            if (overriddenTasks.TryGetValue(taskName, out RegisteredTaskRecord rec))
```

Comment:
Note: I moved this check down 1 layer from `GetRegisteredTask` to `GetTaskRegistrationRecord` since the former calls the latter no matter what.

Unit tests also call directly into the latter.